### PR TITLE
objects with no files should complete the pre-assembly job successfully

### LINKED ISSUE #998: Objects with no files should not put pre-assembly into an error state
Pre-assembly jobs that have an no files should not record an error and should finish as "completed", not "completed (with errors)" state.  They will continue to be marked as "success" for the object in the YAML file (current behavior).

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made? 🤔

Fixes #998 - when a pre-assembly job runs and an object has no files, this is marked as "success" (current behavior), but should not put the job run into a "completed (with errors)" state, but instead should just be "completed" (new behavior).

Update the recently added integration test to verify these jobs end in a completed state with no error.
~~Wait until #997 is merged, rebased here and then updated to reflect this new behavior~~

## How was this change tested? 🤨

Updated existing test from #997 (that new test started failing with my code change here until I adjusted the test)

### CODE REVIEW

File: app/lib/pre_assembly/batch.rb [hunk 1]
```diff
@@ -79,6 +80,7 @@ def stager
       staging_style_symlink ? LinkStager : CopyStager
     end
 
+    # any objects that have already been run successfully through a pre-assembly job
     def pre_assembled_object_containers
```
Comment (peetucket) [inline app/lib/pre_assembly/batch.rb:84]: some additional code comments

File: app/lib/pre_assembly/batch.rb [hunk 2]
```diff
@@ -67,7 +68,7 @@ def digital_objects
     end
     # rubocop:enable Metrics/AbcSize
 
-    # used by discovery report
+    # any objects that have not yet been run successfully through a pre-assembly job (used by both pre-assembly and discovery reports)
     # @return [Array<PreAssembly::DigitalObject>]
```
Comment (peetucket) [inline app/lib/pre_assembly/batch.rb:72]: some additional code comments

File: app/lib/pre_assembly/batch.rb (hunk 1 above)
Comment (ndushay) [reply app/lib/pre_assembly/batch.rb:84]: thank you!

File: app/lib/pre_assembly/batch.rb [hunk 3]
```diff
@@ -67,7 +68,7 @@ def digital_objects
     end
     # rubocop:enable Metrics/AbcSize
 
-    # used by discovery report
+    # any objects that have not yet been run successfully through a pre-assembly job (used by both pre-assembly and discovery reports)
```
Comment (ndushay) [inline app/lib/pre_assembly/batch.rb:71]: thanks for correcting this!

**Review by ndushay — APPROVED** [review]

**Pull request merged**